Feat/log in view

nattsnb opened the pull request.

Commit:
fixed banner styling and displaying  conditions. added useLocalization to provide porper components display depending on path

Commit:
added config directory with bannerConfig.ts, getBannerConfig function in utils to provide the confid, updated Banner component to display the elemnt according to the provided config

Commit:
if provided filters in any other location than explore venue view, on submit will navigate to '/' path

Commit:
added basic view composition with placeholders, provided picture context, added useVenueDetailedView.ts

Commit:
added calendar for booking the venue and fixed styles in files

Commit:
updated venueDto, added ContactInfo section

Commit:
added DetailsAndImageContainer, added fixed number of reviews

Commit:
added LinkBarAndBody component

nattsnb committed:
style(SearchBar/DataInput): removed unnecessary argument

nattsnb committed:
style( DetailsAndImageContainer): use uuid instead of index for mapping. added uuid package
fix(VenueDetailedView) added component name instead of placeholder, must have changed conditions to provide venue exists in both cases

nattsnb committed:
feat(Gallery) added gallery component

nattsnb committed:
feat(Gallery) added gallery component

nattsnb committed:
fix(VenueDetailedView) provided activeVenue with context

nattsnb committed:
fix(Gallery.styled) deleted unneeded type
feature(ExploreVenuesViews & Banner) moved buttons to toggle drawers to banner component, deleted double displayed banner. fixed styling
fix(SearchBar.styled) deleted muted and unneeded code

nattsnb committed:
style(Calendar) moved full prise calculations to useCalendar.ts
feat(DetailsAndImageContainer) provided number of stars and reviews based on data from server
fix(types) updated types according to backend

nattsnb committed:
feat(theme) added new colors to background palette
fix(styles) fixed theme interpolation nad prop types

nattsnb committed:
fix(NavBar) deleted inline styling

nattsnb committed:
after merge

nattsnb committed:
fix(SearchBar) fixed styling

prettier

nattsnb committed:
bug(all) flickering app while starting
fix(Gallery) deleted doubled component mounting
fix(Gallery&Map) provided screen width with prop not to use useMediaQuery multiple times
fix(VenueDetailedView) deleted doubled component mounting
(todo) provide is hidden prop to proper components while in mobile view

nattsnb committed:
bug(all) flickering app while starting
feat(VenueSection) component that will allow to generate most of the components only once
fix(WideBodyLinkBarAndContentContainer) doesn't generate the components anymore
style(VenueDetailedView) used css to hide components

nattsnb committed:
fix(VenueDetailedView) fix the flickering issue - no dependency in useEffect, reordered the initial checks

prettier

nattsnb committed:
fix(VenueDetailedView) fix the flickering issue - no dependency in useEffect, reordered the initial checks

prettier

nattsnb committed:
fix(VenueDetailedView) added missing dependencies

nattsnb committed:
feat(VenueSections) added styled div to provide border in desktop view and central alignment in mobile view

nattsnb committed:
fix(Calendar) fixed props name spelling to avoid mui warnings

nattsnb committed:
feat(VenueDetailedView) added displayedSection to ActiveVenue context to provide easier acces to data

nattsnb committed:
feat(VenueDetailedView) added drawer that display calendar in mobile view
fix(Calendar) added extra styling for calendar to display properly

nattsnb committed:
fix(Layout) deleted unneeded import
fix(Gallery,Map) isMobile is taken from hook now
fix(VenueDetailedView) div fixed to bottom is over map now

nattsnb committed:
fix(all) changed variables names to isMobile when there's change in display dependent on it
fix(Gallery, Map, VenueSections) changed wrong condition in useMediaQuery hook

nattsnb committed:
feat(theme) added new color
fix(Calendar) fixed buttons and calendar styling

nattsnb committed:
feat(theme) added new color

nattsnb committed:
fix(Calendar) fixed code formatting and changed empty fragment into null

nattsnb committed:
feat(VenueDetailedView) used IntersectionObserver to make narrowNavBar stick to the top of mobile view

nattsnb committed:
fix(all) the app returns null instead of empty fragment

nattsnb committed:
fix(Calendar) fix styling of "starts at" "ends at" buttons
fix(theme) deleted unneeded background color
fix(Banner) sort and filter buttons are now displayed only in one location

nattsnb committed:
fix(SearchBar) added FormController and used it for GuestsNumberInput

nattsnb committed:
fix(all) added form controller for all inputs in SearchBar

fix(DateInput) deleted unneeded useDateInput file

nattsnb committed:
merged branch feat/explore-venues-view

nattsnb committed:
fix(NavBar) changed styling of link in mobile view

nattsnb committed:
fix(NavBar) changed name of variable

nattsnb committed:
fix(FiltersDrawer) fixed handleDropClick function

fix(ResultsList) app displays 'No venues found' if venuesOnPage.length is 0, deleted console.logs

nattsnb committed:
fix(DateInput) buttons starts at and ends at are properly styled and they change their appearance depending on which date is to be picked

nattsnb committed:
fix(SearchBar) changed labels into placeholders in SearchBar inputs, styled I want to be more specific button,
feat(theme) provided inputs not to change color when focused on
feat(readme) added read me

nattsnb committed:
fix(LocalizationInput) autoComplete is off

nattsnb committed:
feat(BackButton) Back Button is separate element now

nattsnb committed:
feat(LoginView) started creating loginView. provided state variable to define which mode is active login or register, and building different app layouts depending on that

nattsnb committed:
fix(types) fix types for user and authentication, phonenumber and name are optional, password is not returned in the response.

feat(authenticationApi) added api file for authentication endpoints

feat(LoginView) started building forms for login and user registration

nattsnb committed:
fix(types) deleted login-form-dto, moved register-form-dto to seperate file, typo logIn to login
feat(all) installed axios package and https.ts calling axios instance

nattsnb committed:
chore(http) renamed http to axiosClient
chore(api) used axiosClient for backend calls

nattsnb committed:
feat(authenticationApi) added logoutUser and getAuthenticatedUser functions

nattsnb committed:
feat(contexts) added authenticationContext with functions

chore(all) prettier

nattsnb committed:
feat(LoginView) used SubmitHandler for sumbimitting the forms, added password matching check and provided TextFields for the form

chore(AuthenticationContext) renamed useAuthenticationContext to useAuthentication

chore(all) prettier

nattsnb committed:
feat(AccountView) created new view with user details and reservations

chore(LoginView) fixed styling

nattsnb committed:
feat(AccountView) added to section my reservations

feat(ReservationApi) added reservation api

nattsnb committed:
feat(LoginView)
 - moved google logo image src to variable
 - error is not set if user logs in
 - added a div holding space for possible error message
 - fixed button styling for them not to disappear while hovered over or in focus

nattsnb committed:
feat(AccountView)
 - added useAccountView and moved logic in there
 - produce data with reservation and venue details
 - added component ReservationCard to display reservations
 - added myReservationDto

 feat(LoginView)
 not saved changes from previous commit, styling buttons

nattsnb committed:
feat(ReservationCard) added component structure and basic styling

feat(AccountView) added container for styling

feat(VenueCard) added minimal width

nattsnb committed:
feat(ReservationCard) added styling

nattsnb committed:
feat(NavBar) if user is logged in, the avatar shows up in the NavBar

nattsnb committed:
feat(bannerConfig) added config for account page

feat(VenueCard) added max width

feat(AccountView) added classes for sections to use in styling

feat(ReservationCard) fix styling

nattsnb committed:
feat(Calendar) margins shrinks if there is 6 weeks in a month to prevent scrollbar appearing

nattsnb committed:
feat(Calendar) added functionality to book button

nattsnb committed:
feat(VenueDetailedView) deleted inline styles

nattsnb committed:
feat(Description) deleted unneeded type

feat(Gallery) changed image key

nattsnb committed:
feat(FormController) changed to generic one

feat(SearchBar) deleted unneeded types

nattsnb committed:
merge

Solnick commented on src/components/Banner/SearchBar/LocalizationInput/index.tsx:
```diff
@@ -0,0 +1,57 @@
+        <StyledSuggestionList>
+          {suggestions.map((name, idx) => (
+            <ListItem
+              key={idx}
```
please do not use indexes as key

nattsnb replied:
done

Solnick commented on src/components/VenueDetailedView/Gallery/index.tsx:
```diff
@@ -0,0 +1,42 @@
+      <StyledGalleryContainer>
+        {activeVenue.images.map((imageURL, index) => (
+          <StyledImageContainer
+            key={uuidv4()}
```
don't change key with every rerender

nattsnb replied:
done

Solnick commented on src/App.tsx:
```diff
@@ -1,61 +1,104 @@
+        >
+          <CurrencyProvider>
+            <AuthenticationProvider>
+              <FilterParamsContext.Provider
```
Be consistent and use one notation for context provider. Preferably like this `AuthenticationProvider`

nattsnb replied:
done

Solnick commented on src/App.tsx:
```diff
@@ -1,61 +1,104 @@
+                          path="/start-hosting/"
+                          element={<p>start hosting</p>}
+                        />
+                        <Route path="/login/" element={<LoginView />} />
```
this shoudl be unreachable if someone is logged in

nattsnb replied:
done

Solnick commented on src/components/BackButton/index.tsx:
```diff
@@ -0,0 +1,23 @@
+  return (
+    <StyledBackButtonContainer>
+      <StyledBackButtonFlexDiv onClick={() => navigate(-1)}>
+        <StyledArrowBackIosIcon /> <p>{message}</p>
```
this jsx formatting is off

nattsnb replied:
done

Solnick commented on src/components/LoginView/index.tsx:
```diff
@@ -0,0 +1,238 @@
+        message: "Wrong credentials",
+      });
+    } catch (error) {
+      console.error("Login error:", error);
```
remove console logging

nattsnb replied:
done

nattsnb committed:
feat(LocalizationInput) changed list key

nattsnb committed:
Revert "merge"

This reverts commit 9559348dcea79b52b01f2490185f89fd3c43a4f8, reversing
changes made to ec23338db00b95ca69afe47907038e11e93a5dfc.

nattsnb committed:
feat(Gallery) changed keys

nattsnb committed:
feat(BackButton) fixed formating

feat(AuthenticationContext) deleted console.logs

nattsnb committed:
feat(context) provided uniform ContextProvider construction for all the context

feat(BackButton) fixed indentation

chore(all) prettier

Solnick commented on src/components/AccountView/index.tsx:
```diff
@@ -0,0 +1,74 @@
+  const onClickLogoutButton = async () => {
+    try {
+      await logout();
+      navigate("/login/", { replace: true });
```
Please create enum for routes like this:
```typescript
enum Routes {
login = 'login',
venues = 'venues'
}
```
and use it everywhere where you are reffering to a url path

nattsnb replied:
done

Solnick commented on src/components/Banner/SearchBar/index.tsx:
```diff
@@ -61,13 +63,61 @@ export const SearchBar = () => {
+                    />
+                  )}
+                </FormController>
+                <FormController name="occasionIds">
```
I think you don't need those formControllers, you should assign a name to the input itself. Please test if this will work.

Solnick replied:
And when you stop using the controllers the onchange function will also be redundant

nattsnb replied:
moved issue to fix/filter-form

Solnick commented on src/components/LoginView/LoginView.styled.ts:
```diff
@@ -0,0 +1,241 @@
+    width: 220px;
+    height: 78px;
+    text-decoration: ${isactive ? "underline" : "none"};
+      &:focus, &:focus-visible, &:active {
```
formatting

nattsnb replied:
done

Solnick commented on src/components/LoginView/LoginView.styled.ts:
```diff
@@ -0,0 +1,241 @@
+    
+    ${theme.breakpoints.down("md")} {
+      width: 190px;
+    height: 66px;
```
formatting

nattsnb replied:
done

Solnick commented on src/components/VenueDetailedView/VenueSections.tsx:
```diff
@@ -0,0 +1,71 @@
+        <Description descriptionRef={descriptionRef} />
+      </section>
+      <section
+        style={{
```
inline styles 🤮

nattsnb replied:
there was a merge issue, fixed

nattsnb committed:
feat(VenueSections) added StyledSection, fixed merging error

nattsnb committed:
feat(NoUserRoute) created route that prevents logged-in user from entering logging page

feat(ProtectedRoute) fixed arguments

nattsnb committed:
chore(styled) fixed indents

nattsnb committed:
fix(App) fixed wrong enum name

nattsnb committed:
feat(.env) provided env files with api url

nattsnb committed:
feat(index.html) changed title and fav icon

nattsnb committed:
fix(SearchBar) fixed look of search bar

nattsnb committed:
fix(evn) typo

nattsnb committed:
fix(README) update

feat(searchBar) renamed isCollapsed to isSearchBarCollapsed and add it as prop to LocalizationInput

feat(LocalizationInput) added onFocusLocalizationInput function that sets the whole SearchBar open while focused on LocalizationInput

nattsnb committed:
fix(env) changed address
